File: frontend/src/api/stockBasic.ts
import http from './http'

export interface StockBasicItem {
  code: string
  name: string | null
  market: string | null
  industry_name: string | null
  region: string | null
  list_date: string | null
  synced_at: string | null
  data_source: string | null
}

export interface StockBasicListResponse {
  items: StockBasicItem[]
  total: number
  page: number
  page_size: number
  last_synced_at: string | null
}

export function getStockBasicList(params: {
  page?: number
  page_size?: number
  code?: string
  name?: string
  market?: string
  industry?: string
}) {
  return http.get<StockBasicListResponse>('/stock/basic', { params })
}

/** 全市场列表同步可能较慢，单独放宽超时（毫秒）。 */
const SYNC_TIMEOUT_MS = 600_000

export function postStockBasicSync() {
  return http.post<{ status: string; message: string; stock_basic: number }>(
    '/stock/basic/sync',
    {},
    { timeout: SYNC_TIMEOUT_MS }
  )
}

File: backend/app/schemas/stock_basic.py
"""股票基本信息 API 模型。"""
from datetime import date, datetime

from pydantic import BaseModel, Field


class StockBasicItem(BaseModel):
    code: str
    name: str | None = None
    exchange: str | None = None
    market: str | None = None
    industry_name: str | None = None
    region: str | None = None
    list_date: date | None = None
    synced_at: datetime | None = None
    data_source: str | None = None

    model_config = {"from_attributes": True}


class StockBasicListResponse(BaseModel):
    items: list[StockBasicItem]
    total: int
    page: int = Field(ge=1)
    page_size: int = Field(ge=1, le=200)
    last_synced_at: datetime | None = None


class StockBasicSyncResponse(BaseModel):
    """同步完成后返回（请求内同步执行）。"""
    status: str = "ok"
    message: str = "股票基本信息同步完成"
    stock_basic: int = 0
